### DISCUSSION TITLE
auto register views to files on s3 / query file-list on s3

### DISCUSSION THREAD

**Discussion by OneCyrus** [duckdb/duckdb#9179] [General]
i would like to auto register all files (parquet, json) on a S3 bucket in duckdb. is there a native way to e.g. point duckdb to an s3 bucket and it creates views from all files?

or is there a way to run a SELECT against a s3 bucket to return all the files so we could use the returned list for dynamically creating views?

**Comment by carlopi** [comment]
`FROM glob('s3://some-subfolder/**/*.parquet')` should return the list of files in a given bucket.
